R/tmiic.wrapper.R

```diff
@@ -0,0 +1,618 @@
+#   \cr    \tab                \tab                \tab    \tab        \tab                \tab               \cr
+#   t-8    \tab node1_val(t-8) \tab node2_val(t-8) \tab => \tab   i"   \tab node1_lag6_val \tab node2_lag6_val\cr
+#   t-5    \tab node1_val(t-5) \tab node2_val(t-5) \tab => \tab   i"   \tab node1_lag3_val \tab node2_lag3_val\cr
+#   t-2    \tab node1_val(t-2) \tab node2_val(t-2) \tab => \tab   i"   \tab node1_lag0_val \tab node2_lag0_val\cr
+#   \cr    \tab                \tab                \tab    \tab        \tab                \tab               \cr
+#   ...    \tab .............. \tab .............. \tab => \tab ...... \tab .............. \tab ............. \cr
+# }
+# until number of timesteps - \emph{tau} * \emph{delta_tau} is reached. 
+# The same process is applied to all input timeseries.
+#
+# @param input_data [a dataframe] 
+# A dataframe with the time series with variables as columns and
+# timeseries/timesteps as rows. The timestep information must be suplied 
+# in the first column and, for each timeseries, in an ascending order.
+#
+# @param tau [an int > 0] A strictly positive int defining the max lag.
+# Note that if \emph{delta_tau} is also supplied, \emph{tau} must be a
+# multiple of \emph{delta_tau}.
+# 
+# @param state_order [a data frame] Optional, NULL by default. 
+# A data frame giving information about how to order the various states of 
+# categorical variables. This data frame will be lagged as the input data 
+# on \emph{tau} / \emph{delta_tau} layers.
+# 
+# @param movavg [an integer] Optional, -1 by default.\cr
+# When \emph{movavg} is supplied (integer > 1), a moving average 
+# operation is applied to each timeseries.\cr
+# 
+# @param delta_tau [an integer] Optional, 1 by default.\cr
+# When \emph{delta_tau} is supplied (integer > 1), the samples will be 
+# constructed using 1 timestep every \emph{delta_tau} timesteps starting 
+# from the last.\cr 
+# i.e.: on 1000 timesteps with  \emph{tau} = 14 and \emph{delta_tau} = 7, 
+# the timesteps kept for the samples conversion will be 1000, 993, 986 
+# for the first sample, the next sample will use 999, 992, 985 and so on.\cr
+# 
+# @return a list with two elements:
+# \itemize{
+#  \item \emph{input_data:} the samples generated from the timeseries
+#  \item \emph{state_order:} the lagged state_order 
+#  }
+#                                    
+#-----------------------------------------------------------------------------
+tmiic.transform_data_for_miic <- function (input_data, tau, state_order=NULL, 
+                                           movavg=-1, delta_tau=1) {
+  n_rows <- nrow(input_data)
+  n_nodes <- ncol(input_data) - 1
+  list_nodes <- colnames(input_data)[-1]
+  #
+  # Lag the nodes
+  #
+  list_nodes_lagged <- list()
+  for (tau_idx in seq(0,tau,by=delta_tau) )
+    for (node_idx in 1:n_nodes)
+      list_nodes_lagged <- append (list_nodes_lagged, paste (list_nodes[[node_idx]], "_lag", tau_idx, sep="" ) )
```

Comment:
It's ok if you append "_lag" to lagged variables, for print purpose for example. But don't rely on it with regex for operations like finding specific lagged nodes. Instead build a map between each node and their lagged counterparts to be able to calculate the index of lagged nodes based on the index of non-lagged nodes. See the comment for `tmiic.h` for more.